aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1167,4 +1182,74 @@ Tensor& lu_solve_out(Tensor& result, const Tensor& self, const Tensor& LU_data,
   return result;
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ lstsq ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+template <typename scalar_t>
```

Comment:
Add a comment here describing the function's calling convention and briefly describing how it's implemented.